python/private/pypi/whl_library_targets.bzl

```diff
@@ -362,9 +353,19 @@ def _config_settings(dependencies_by_platform, native = native, **kwargs):
             * `@//python/config_settings:is_python_3.{minor_version}`
             * `{os}_{cpu}`
             * `cp3{minor_version}_{os}_{cpu}`
+        dependencies_with_markers: {type}`dict[str, str]` The markers to evaluate by
+            each dep.
+        rules: used for testing
         native: {type}`native` The native struct for overriding in tests.
         **kwargs: Extra kwargs to pass to the rule.
     """
+    for dep, expression in dependencies_with_markers.items():
+        rules.env_marker_setting(
+            name = "include_{}".format(dep),
```

Comment:
I was expecting to a list of extras passed in. But I guess the extras are being handled earlier?

The case I had in mind was a more complicated boolean expression e.g. `(extra == foo and python_version ~= 3.10) or (extra = bar and python_version ~= 3.9)`; something that requires the target config to be evaluated.

That seems rare, though? And can probably be ignored?

Reply:
Let me think about it a little more.

Reply:
Realized that this should be still fine as is, at least for your example. Leaving the comment unresolved in case something comes along.